packages/teams-js/test/public/mail.spec.ts

```diff
@@ -0,0 +1,325 @@
+          ),
+        );
+    });
+
+    it('should not allow calls from sidePanel context', async () => {
+      await utils.initializeWithContext('sidePanel');
+
+      await mail
+        .composeMail(composeMailParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "sidePanel".'),
+          ),
+        );
+    });
+
+    it('should not allow calls from stage context', async () => {
+      await utils.initializeWithContext('stage');
+
+      await mail
+        .composeMail(composeMailParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "stage".'),
+          ),
+        );
+    });
+
+    it('should not allow calls from meetingStage context', async () => {
+      await utils.initializeWithContext('meetingStage');
+
+      await mail
+        .composeMail(composeMailParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "meetingStage".'),
+          ),
+        );
+    });
+
+    it('should not allow calls if runtime does not support mail', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: {} });
+
+      await mail.composeMail(composeMailParams).catch(e => expect(e).toBe('Not Supported'));
+    });
+
+    it('should successfully throw if the openMailItem message sends and fails', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { mail: {} } });
+
+      const composeMailPromise = mail.composeMail(composeMailParams);
+
+      const composeMail = utils.findMessageByFunc('mail.composeMail');
+
+      const data = {
+        success: false,
+        error: 'Something went wrong...',
+      };
+
+      utils.respondToMessage(composeMail, data);
+      await composeMailPromise.catch(e => expect(e).toMatchObject(new Error('Something went wrong...')));
+    });
+
+    it('should successfully send the openMailItem message', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { mail: {} } });
+
+      const promise = mail.composeMail(composeMailParams);
+
+      const composeMailMessage = utils.findMessageByFunc('mail.composeMail');
+
+      const data = {
+        success: true,
+        error: 'Something went wrong...',
+      };
+
+      utils.respondToMessage(composeMailMessage, data);
+      await promise;
+
+      expect(composeMailMessage).not.toBeNull();
+      expect(composeMailMessage.args.length).toEqual(1);
+      expect(composeMailMessage.args[0]).toStrictEqual(composeMailParams);
```

Comment:
Same comment as on line 162 about responding to the message being unnecessary?

Reply:
Repeating here just in case: believe you have to respond to the message or the promise never resolves and the test times out after 5 seconds.